core/PaginatedList.php

```diff
@@ -229,9 +229,19 @@ public function Pages($max = null) {
 		}
 
 		for ($i = $start; $i < $end; $i++) {
+
+			// Remove the query sting for the first page link
+			// /?start=0 returns the same page as /
+			$link = HTTP::setGetVar($this->getPaginationGetVar(), $i * $this->getPageLength());
+
+			if($i === 0) {
+				$parts = explode("?", $link);
```

Comment:
Won't this remove ALL querystring args? What if there's another arg that is needed to render.

E.g.

`http://www.mysite.com/page?show=category1&start=0` should be changed into `http://www.mysite.com/page?show=category1` not `http://www.mysite.com/page`